src/components/helpers/dom_dnd_hook.ts

```diff
@@ -183,15 +278,24 @@ export class DOMDndHelper {
       }
 
       let mouseOffset = this.currentMousePosition - this.initialMousePosition;
-      let left = mouseOffset;
-      left = Math.max(this.minPosition - item.positionAtStart, left);
-      left = Math.min(this.maxPosition - item.positionAtStart - item.size, left);
+      let start = mouseOffset;
+      start = Math.max(this.minPosition - item.positionAtStart, start);
+      start = Math.min(this.maxPosition - item.positionAtStart - item.size, start);
 
-      positions[item.id] = left;
+      positions[item.id] = start;
     }
+
     return positions;
   }
 
+  private get containerStart() {
+    return this.direction === "horizontal" ? this.containerRect.left : this.containerRect.top;
```

Comment:
instead of if branches here and there, we could also have two very small wrapper classes `HorizontalContainer` and `VerticalContainer` to abstract away the differences from the main drag&drop logic